core/dbt/parser/manifest.py

```diff
@@ -841,11 +845,7 @@ def build_public_nodes(self) -> bool:
 
     def load_new_public_nodes(self):
         for project in self.manifest.project_dependencies.projects:
-            publication = (
-                next(p for p in self.publications if p.project_name == project.name)
-                if self.publications
-                else None
-            )
+            publication = self.publications.get(project.name)
```

Comment:
What about putting the `raise` closer to the cause of the exception? For example:

```python
try:
    publication=self.publications[project.name]
except KeyError:
    raise PublicationConfigNotFound(project=project.name)

publication_config = PublicationConfig.from_publication(publication)
self.manifest.publications[project.name] = publication_config

for public_node in publication.public_models.values():
    self.manifest.public_nodes[public_node.unique_id] = public_node
```

Reply:
I like it! will update